src/zc/buildout/buildout.py

```diff
@@ -1172,11 +1172,11 @@ def _maybe_upgrade(self):
 
         # Restart
         args = sys.argv[:]
-        args.insert(1, '--restart-after-upgrade')
         if not __debug__:
             args.insert(0, '-O')
         args.insert(0, sys.executable)
-        sys.exit(subprocess.call(args))
+        env = {'BUILDOUT_RESTART_AFTER_UPGRADE': '1'}
```

Comment:
Shouldn't this be something like 
```python
        env = dict(os.environ, BUILDOUT_RESTART_AFTER_UPGRADE='1')
```
to also preserve environment from current process variables ?

Reply:
@perrinjerome Right, I missed that.